Yoast/YoastSEO.js
# 1773 worker runresearch failsafe

## Summary

This PR can be summarized in the following changelog entry:

* The browser console now shows descriptive error messages when triggering single researches via the webworker.

## Relevant technical choices:

* The webworker returns an `runResearch:failed` message with a descriptive error message, but this error message is not handled yet in the plugin. In the mean time we log the error to the console.

## Test instructions

This PR can be tested by following these steps:

* Open a research that is called using the `runResearch`-method of the webworker in PHPStorm or another editor.
    * Like the `relevantWords` or `getLinks` researches.
* Edit the code in such away that it throws an error.
   * E.g. add
      ```js
      throw new Error( "This is an error" );
      ```
      In `relevantWords` or `getLinks` before the `return` statement.
* Open a post in WordPress with the premium plugin activated.
* Open the browser console.
* Notice that it shows a descriptive error in the console, with the given error message as defined above.
    * Error should be: 
       ```
        An error occurred after running the 'relevantWords' research.
	      Error: This is an error
       ```
    * Note that is shows an extra error (`Uncaught (in promise)`), since the error is returned from the webworker as well, but not handled yet by the plugin.

Follow the same steps as above, but:
* Edit the code such that it throws a random object.
    ```js
     throw { someData: 42 }
    ```
* The error in the console should read:
   ```
   An error occurred after running the 'relevantWords' research.
   ```
 
Fixes #1773
